wagtail/admin/tests/test_pages_views.py

```diff
@@ -2102,6 +2105,41 @@ def hook_func(request, page):
         # page should be edited
         self.assertEqual(Page.objects.get(id=self.child_page.id).title, "I've been edited!")
 
+    def test_construct_page_action_menu_hook(self):
+        def hook_func(menu_items, request, context):
+            menu_items[:] = [
+                UnpublishMenuItem(order=10),
+                DeleteMenuItem(order=20),
+                SubmitForModerationMenuItem(order=30),
+                SaveDraftMenuItem(order=40),
+                PublishMenuItem(_isDefault=True),
+            ]
+
+        default_item = '<button type="submit" name="action-publish" value="action-publish" class="button button-longrunning " data-clicked-text="Publishing…"><span class="icon icon-spinner"></span><em>Publish</em></button>'
+        menu_items_html = '''
+        <ul>
+            <li>
+                <a class="button" href="/admin/pages/6/unpublish/">Unpublish</a>
+            </li>
+            <li>
+                <a class="button" href="/admin/pages/6/delete/">Delete</a>
+            </li>
+            <li>
+                <input type="submit" name="action-submit" value="Submit for moderation" class="button" />
+            </li>
+            <li>
+                <button type="submit" class="button action-save button-longrunning " data-clicked-text="Saving…" ><span class="icon icon-spinner"></span><em>Save draft</em></button>
+            </li>
+        </ul>
+        '''
+        with self.register_hook('construct_page_action_menu', hook_func):
+            response = self.client.get(reverse('wagtailadmin_pages:edit',
+                                       args=(self.single_event_page.id, )))
+
+        self.assertContains(response, default_item, html=True)
+        self.assertContains(response, menu_items_html, html=True)
+
+
```

Comment:
Also need to test:
 - Not specifying `default`
 - Using `default` more than once

Reply:
I change functionality to last menu_item is default.